### PR TITLE
Improves VSCode Launch Configuration

### PR DESCRIPTION
**Pull request by zoharcochavi**
Improves vscode launch config to work with default cpp extension, and… launches debugger in one task

### CODE REVIEW

**Review by jonay2000 — APPROVED** [review]
looks good!

**Comment by ValentijnvdBeek** [comment]
Looks good, maybe someone who uses VSCode could verify it but it is small enough of a change that I feel we can just merge it

**Comment by jonay2000** [comment]
That's what I thought

**Comment by zoharcochavi** [comment]
I tagged @Dimitar2000 since he's also using VSCode! @jonay2000 do you know if @NULLx76 also uses vscode and still wants to review? He was the last person to work on this.

**Comment by jonay2000** [comment]
He does use it, but for course os he used clion like me.

File: .vscode/launch.json [hunk 1]
```diff
@@ -2,16 +2,24 @@
...
-            "remote": true,
-            "cwd": "${workspaceRoot}",
-            // "preLaunchTask": "run_debug",
-            "valuesFormatting": "prettyPrinters"
+            "name": "(gdb) Attach",
+            "type": "cppdbg",
+            "request": "launch",
+            "program": "${workspaceFolder}/src/kernel/build/kernel.sym",
+            "cwd": "${workspaceFolder}/src/kernel/",
+            "miDebuggerPath": "${workspaceFolder}/src/toolchain/arm-none-eabi/bin/arm-none-eabi-gdb",
```
Comment (NULLx76) [inline .vscode/launch.json:10]: I have `arm-none-eabi-gdb` installed through my package manager so this configuration doesn't work as it complains the path doesn't exist. Maybe it's possible to add `${workspaceFolder}/src/toolchain/arm-none-eabi/bin/arm-none-eabi-gdb` to the runners `PATH` and then just specify `arm-none-eabi-gdb` here?

**Comment by Dimitar2000** [comment]
> I tagged @Dimitar2000 since he's also using VSCode! @jonay2000 do you know if @NULLx76 also uses vscode and still wants to review? He was the last person to work on this.

File: .vscode/launch.json (hunk 1 above)
Comment (zoharcochavi) [reply .vscode/launch.json:10]: Good idea, will do!

**Comment by zoharcochavi** [comment]
I will also add and readjust the other configuration, as the open-source version of VSCode doesn't support the native C/C++ extension.
In this case you should use [Native Debug](https://marketplace.visualstudio.com/items?itemName=webfreak.debug)

**Comment by zoharcochavi** [comment]
So currently it launches, attaches, and stops automagically in both setups (closed- and open-source).
However:

1. The closed-source launch config, however, currently requires `sudo` to kill the process.
If anyone has a better solution, feel free to change it!

2. As for setting the path variable, VSCode is a bit whiny when it comes to environment variables. 
I currently have two attempts at changing it via tasks, both of which don't work.

At this point, I've spent more time on this than I would like to admit, so when we're done discussing *1*, I say we close this PR.

**Review by Dimitar2000 — APPROVED** [review]

**Comment by Dimitar2000** [comment]
Currently I do not use VSCode for debugging, but I tested that it works on my laptop too, so if I decide to switch to VSCode debugging, I will know it works. If the configuration works for everyone else, that should be good enough.

**Pull request merged**